### PR TITLE
[Backport 2.x] [Tiered Caching] Serializers for ehcache (#12709)

### PR DESCRIPTION
**Pull request by peteralfonsi**
**Original PR: https://github.com/opensearch-project/OpenSearch/pull/12709**

Signed-off-by: Peter Alfonsi <petealft@amazon.com>
Co-authored-by: Peter Alfonsi <petealft@amazon.com>
(cherry picked from commit 21b28f2a019a8ee538a763453366e6fd4943b768)

### Description
As part of tiered caching, adds a Serializer interface and implementations for BytesReference and IndicesRequestCache.Key. Also integrates these with EhcacheDiskCache. This is required for Ehcache to accept non-primitive keys or values. 

### Related Issues
Part of the [tiered caching feature. ](https://github.com/opensearch-project/OpenSearch/issues/10024)

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
~- [N/A] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by VachaShah — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - [Metrics Framework] Adds support for asynchronous gauge metric type. ([#12642](https://github.com/opensearch-project/OpenSearch/issues/12642))
 - Make search query counters dynamic to support all query types ([#12601](https://github.com/opensearch-project/OpenSearch/pull/12601))
 - [Tiered caching] Add policies controlling which values can enter pluggable caches [EXPERIMENTAL] ([#12542](https://github.com/opensearch-project/OpenSearch/pull/12542))
+- [Tiered caching] Add Stale keys Management and CacheCleaner to IndicesRequestCache ([#12625](https://github.com/opensearch-project/OpenSearch/pull/12625))
```
Comment (andrross) [inline CHANGELOG.md:26]: This looks unrelated to this PR?

Comment (peteralfonsi) [reply CHANGELOG.md:26]: This must have come through when resolving the merge conflict when this was added to main - I'll remove it from the backport branch.

**Comment by peteralfonsi** [comment]
https://github.com/opensearch-project/OpenSearch/issues/12743

**Review by andrross — APPROVED** [review]
